src/Stratis.Features.FederatedPeg/Wallet/FederationWalletManager.cs

```diff
@@ -303,6 +303,27 @@ public void RemoveBlocks(ChainedHeader fork)
             }
         }
 
+        /// <summary>
+        /// Determines if the wallet is active.
+        /// The wallet will only become active after <see cref="FederationWallet.LastBlockSyncedHeight"/>.
+        /// </summary>
+        /// <param name="height">The height at which to test if the wallet should be active. Defaults to the chain indexer height.</param>
+        /// <returns></returns>
+        private bool IsWalletActive(int? height = null)
+        {
+            if (this.Wallet == null)
+                return false;
+
+            if (this.Wallet.LastBlockSyncedHash != null)
+                return true;
+
+            if (this.Wallet.LastBlockSyncedHeight >= (height ?? this.chainIndexer.Height))
+                return false;
+
+            this.Wallet.LastBlockSyncedHash = this.chainIndexer[(int)this.Wallet.LastBlockSyncedHeight].HashBlock;
```

Comment:
I think that a method called `IsWalletActive` should not change state its missleading. otherwise call it `IsWalletActiveThenUpdateWalletHeight`

Reply:
Anywy the only place to update the wallet heigh is in `ProcessesBlock` so this change should not happen here IMO but outside of the method

Reply:
Changed.